torch_frame/nn/models/excelformer.py

```diff
@@ -152,79 +188,101 @@ def __init__(
         self.excelformer_decoder = ExcelFormerDecoder(in_channels,
                                                       out_channels, num_cols)
         self.reset_parameters()
+        self.mixup = mixup
+        self.beta = beta
 
     def reset_parameters(self) -> None:
         self.excelformer_encoder.reset_parameters()
         for excelformer_conv in self.excelformer_convs:
             excelformer_conv.reset_parameters()
         self.excelformer_decoder.reset_parameters()
 
-    def forward(self, tf: TensorFrame) -> Tensor:
+    def forward(self, tf: TensorFrame,
+                mixup_encoded: bool = False) -> Tensor | tuple[Tensor, Tensor]:
         r"""Transform :class:`TensorFrame` object into output embeddings.
 
         Args:
             tf (:class:`torch_frame.TensorFrame`):
                 Input :class:`TensorFrame` object.
+            mixup_encoded (bool):
+                Whether to mixup on encoded numerical features, i.e.,
+                FEAT-MIX and HIDDEN-MIX.
 
         Returns:
             torch.Tensor: The output embeddings of size
                 [batch_size, out_channels].
         """
-        if stype.numerical not in tf.feat_dict:
-            raise ValueError(
-                "Excelformer only takes in numerical features, but the input "
-                "TensorFrame object does not have numerical features.")
         x, _ = self.excelformer_encoder(tf)
+        # FEAT-MIX or HIDDEN-MIX is compatible with `torch.compile`
+        if mixup_encoded:
+            assert tf.y is not None
+            x, y_mixedup = feature_mixup(
+                x,
+                tf.y,
+                num_classes=self.out_channels,
+                beta=self.beta,
+                mixup_type=self.mixup,
+                mi_scores=getattr(tf, 'mi_scores', None),
+            )
```

Comment:
Where did you set `mixup_encoded` to be `True`?

Reply:
I have organized this in a more clear manner. Your mentioned `mixup_encoded = True` appeared in the `forward_mixup` function since I wanted to call `feature` & `hidden` mixup (mixup on encoded features) through `forward_mixup` API. Now I have re-organized the calling style, `forward_mixup` is removed (since we don't need `ordinary` mixup), and the user directly uses mixup for training by simply calling `model(tf, mixup_encoded=True)`.